osu.Game/Screens/Edit/Compose/Components/SelectionHandler.cs

```diff
@@ -237,6 +226,13 @@ internal void HandleDeselected(SelectionBlueprint blueprint)
         /// <param name="blueprint">The blueprint.</param>
         /// <param name="state">The input state at the point of selection.</param>
         internal void HandleSelectionRequested(SelectionBlueprint blueprint, InputState state)
+        {
+            shiftClickDeleteCheck(blueprint, state);
+            multiSelectionHandler(blueprint, state);
```

Comment:
This is not the same as what my suggestion had - that's not how `return` works. This way both methods will be called in their entirety and `multiSelectionHandler` will presumably crash due to having the hitobject/blueprint deleted from under it.

Reply:
I think I understand. Let me just make sure.

If the hitobject/blueprint is deleted in `shiftClickDeleteCheck`, we want the function to return so that `multiSelectionHandler` does not run and crash. Is that correct?

Reply:
That's correct. If the methods are to be split as you have them, the way this is sometimes done is by having the first function return a `bool` that returns if anything was indeed deleted guarding the second call, as so:

```csharp
if (!blueprintDeletionRequested(blueprint, state))
    handleMultiSelection(blueprint, state);
```

Reply:
Okay. I think I understand. Let me just double check.

We want the function `HandleSelectionRequested` to return after running `shiftClickDeleteCheck` if the hitobject/blueprint is deleted. This is so `multiSelectionHandler` doesn't run and isn't confused as to why there is not a hitobject/blueprint.

I think the easiest way to go about this would be to put the code you suggested earlier directly into `HandleSelectionRequested` above where `multiSelectionHandler` is called. Is that okay?

Reply:
Sorry, deleted the old comment by accident. I like your solution. Will implement it now. Thank you so much for your help!